tests/ValidatorDelegates/CSPBadFlagsTest.php

```diff
@@ -0,0 +1,182 @@
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>default-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-inline\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    ),
+                    new Error(
+                        'Content Security Policy contains the
+                        <b>\'unsafe-eval\'</b> keyword in
+                        <b>script-src</b>, which prevents CSP protecting
+                        against the injection of arbitrary code into
+                        the page.',
+
+                        E_USER_WARNING
+                    )
+                ],
+            ],
+            [
+                "default-src 'unsafe-inline' 'unsafe-eval'; "
+                . "script-src 'unsafe-inline' 'unsafe-eval'",
+                [
+                    new Error(
```

Comment:
We'd have to modify the `Error` class for each new error message? Seems to go against the responsability of the class IMO (since specifics of the message are to do with the thing that creates it)?

Something like:
```php
$Error = Error::createBadFlagError($friendlyHeader, $badFlag, $attributeName);
```
?

Perhaps `Error` subtypes might be more fitting?